lib/index.test.js

```diff
@@ -0,0 +1,182 @@
+      extensions: '.js',
+    }));
+
+  it('should still resolve absolutely', async () => {
+    await expect(run(`${fixtures}/flat/main.js`)).resolves.toBe('no imports');
+  });
+
+  it('should not try to import relative to itself', async () => {
+    // We're depending reasonable, yet implicit, behavior: we expect rollup to fail on a missing
+    // entry.
+    await expect(run('/main.js')).rejects.toThrow(/^Could not (?:load|resolve)/);
+  });
+
+  it('should not import from a bad location', () =>
+    expectRejects(
+      run('basic/main.js', {
+        root: `${fixtures}/doesnotexist`,
+      }),
+      (err) => {
+        // Since it doesn't exist, rollup will generate a require call to try and
+        // load the module at runtime. Our require implementation throws an error
+        // and we'll check for that here.
+        expect(err).toBeInstanceOf(Error);
+        expect(err).toHaveProperty('message', 'module was not resolved');
+        expect(err).toHaveProperty('module', '/dir/thing.js');
+      }
+    ));
+
+  it('should not import without extension', () =>
+    expectRejects(run('extension/main.js'), (err) => {
+      expect(err).toBeInstanceOf(Error);
+      expect(err).toHaveProperty('message', 'module was not resolved');
+      expect(err).toHaveProperty('module', '/thing');
+    }));
+
+  it('should try roots in order', () =>
+    Promise.all([
+      expect(
+        run('basic/main-root.js', {
+          root: [`${fixtures}/basic`, `${fixtures}/basic/dir`],
+        })
+      ).resolves.toBe('wrong file!'),
+
+      expect(
+        run('basic/main-root.js', {
+          root: [`${fixtures}/basic/dir`, `${fixtures}/basic`],
+        })
+      ).resolves.toBe('sample'),
+    ]));
+
+  it('should prepend the entry root', () =>
+    Promise.all([
+      expect(
+        run('basic/main-root.js', {
+          useInput: 'prepend',
+          root: `${fixtures}/basic/dir`,
+        })
+      ).resolves.toBe('wrong file!'),
+
+      // This fixture contains its own expectation.
+      run('basic/main-hidden.js', {
+        useInput: 'prepend',
+        root: `${fixtures}/basic/dir`,
+      }),
+    ]));
+
+  it('should append the entry root', () =>
+    Promise.all([
+      expect(
+        run('basic/main-root.js', {
+          useInput: 'append',
+          root: `${fixtures}/basic/dir`,
+        })
+      ).resolves.toBe('sample'),
+
+      // This fixture contains its own expectation.
+      run('basic/main-hidden.js', {
+        useInput: 'append',
+        root: `${fixtures}/basic/dir`,
+      }),
+    ]));
+
+  it('should import with extension', () =>
+    expect(
+      run('extension/main.js', {
+        extensions: '.js',
+      })
+    ).resolves.toBe('no-extra'));
+
+  it('should try extensions in order', () =>
+    expect(
+      run('extension/main.js', {
+        extensions: ['.js', '.js.js'],
+      })
+    ).resolves.toBe('no-extra'));
+
+  it('should try extensions in order', () =>
```

Comment:
Was this supposed to have the same title as the case above? It makes sense for both, just wondering

Reply:
:man_shrugging: these were the existing test names - I don't see much of a reason to make them different, though